solara/server/starlette.py

```diff
@@ -576,6 +576,12 @@ def on_startup():
 
 
 def on_shutdown():
+    # shutdown all kernels
+    for context in list(kernel_context.contexts.values()):
+        try:
+            context.close()
+        except:  # noqa
+            logger.exception("error closing kernel on shutdown")
```

Comment:
Would it be useful to identify which kernel's shutdown failed for logging purposes?

Reply:
I don't think the ID gives much information, but this will show the stacktrace, which should give a hint.